compiler/src/dotty/tools/dotc/transform/UnusedDecls.scala

```diff
@@ -40,10 +40,11 @@ class ErasedDecls extends MiniPhase with InfoTransformer {
 
   /* Info transform */
 
-  override def transformInfo(tp: Type, sym: Symbol)(implicit ctx: Context): Type = tp match {
-    case tp: ClassInfo =>
-      if (tp.classSymbol.is(JavaDefined) || !tp.decls.iterator.exists(_.is(Erased))) tp
-      else tp.derivedClassInfo(decls = tp.decls.filteredScope(!_.is(Erased)))
-    case _ => tp
+  override def transformInfo(tp: Type, sym: Symbol)(implicit ctx: Context): Type = {
+    if (sym.is(JavaDefined) || sym.is(Scala2x)) tp
```

Comment:
You can use InfoTransformer#mayChange instead, it's more efficient because it avoids having to force the info of the symbol: https://github.com/lampepfl/dotty/blob/2b342722681bf49780e575189356b0013f4dbfe6/compiler/src/dotty/tools/dotc/core/DenotTransformers.scala#L55-L59